python_katas/kata_1/test.py

```diff
@@ -29,102 +27,222 @@ class TestWordsConcatenation(unittest.TestCase):
+
 
 
 class TestPairMatch(unittest.TestCase):
     """
     3 Katas
     """
 
-    def test_sample(self):
-        # your code here
-        pass
+    def negative_vaules(self):
+        men = {"John": -20, "Abraham": -45}
+        women = {"July": -18, "Kim": -26}
+        result = ('John', 'July')
+        self.assertEqual(questions.pair_match(men, women), result)
+
+    def is_result_tuple(self):
+        men = {"John": -20, "Abraham": -45}
+        women = {"July": -18, "Kim": -26}
+        self.assertIsInstance(questions.pair_match(men,women), tuple)
+
+    def none_value(self):
+        men = {"John": -20, "Abraham": -45}
+        women = {"July": None, "Kim": -26}
+        self.assertEqual(questions.pair_match(men,women), ('Abraham', 'Kim'))
 
+    def none_key(self):
+        men = {None: 255, "Abraham": 55}
+        women = {"July": 55, "Kim": 256}
+        self.assertEqual(questions.pair_match(men,women), (None, 'Kim'))
+
+    def zero_values(self):
+        men = {'Mony': 0, "Nick": 0}
+        women = {"Imbal": 0, "Irit": 0}
+        self.assertEqual(questions.pair_match(men,women), ('Mony', 'Imbal'))
 
 class TestBadAverage(unittest.TestCase):
     """
     1 Katas
     """
 
-    def test_sample(self):
-        # your code here
-        pass
+    def test_zeros(self):
+        self.assertEqual(questions.bad_average(0, 0, 0), 0)
+
+    def test_negative(self):
+        self.assertEqual(questions.bad_average(-5, -5, -5), -5)
+
+    def test_correct(self):
+        self.assertEqual(questions.bad_average(20, 40, 40), 33)
+
+    def test_float(self):
+        self.assertEqual(questions.bad_average(5.5, 5.5, 5.5), 5)
+
+    def test_return_int(self):
+        self.assertIsInstance(questions.bad_average(0, 0, 0), int)
 
 
 class TestBestStudent(unittest.TestCase):
     """
     1 Katas
     """
 
-    def test_sample(self):
-        # your code here
-        pass
+    def test_none_value(self):
+        data = {
+        "Ben": 78,
+        "Dan": 88,
+        "Nathan": None,
+        "Daniel": 65,
+        "Tal": 95
+        }
+        self.assertEqual(questions.best_student(data), 'Tal')
+
+    def return_string(self):
+        data = {
+            "Ben": 78,
+            "Dan": 88,
+            "Nathan": None,
+            "Daniel": 65,
+            "Tal": 95
+                }
+        self.assertIsInstance(questions.best_student(data), str)
+
+    def test_float(self):
+        data = {
+            "Ben": 78.55,
+            "Dan": 88.66,
+            "Nathan": None,
+            "Daniel": 65.66,
+            "Tal": 5.1
+        }
+        self.assertEqual(questions.best_student(data), 'Dan')
+
+    def test_value_string(self):
+        data = {
+            None: 'x',
+            "Dan": 88,
+            "Nathan": None,
+            "Daniel": 65,
+            "Tal": 95
+        }
+        self.assertEqual(questions.best_student(data), 'Tal')
+
+    def test_key_not_str(self):
```

Comment:
The function expects `dict` of `str` -> `int` | `float`. So no need to test this case